app/src/org/commcare/preferences/CommCarePreferences.java

```diff
@@ -167,40 +151,39 @@ private void hideServerPrefsIfNeeded() {
         }
     }
 
-    private void setupUI() {
-        setTitle(Localization.get("settings.main.title"));
-        addBackButtonToActionBar(this);
-        setupLocalizedText(this, keyToTitleMap);
-        setupButtons();
-    }
-
-    public static void setupLocalizedText(PreferenceActivity activity,
-                                          Map<String, String> prefToTitleMap) {
-        PreferenceScreen screen = activity.getPreferenceScreen();
-        for (int i = 0; i < screen.getPreferenceCount(); i++) {
-            String key = screen.getPreference(i).getKey();
-            if (prefToTitleMap.containsKey(key)) {
-                try {
-                    String localizedString = Localization.get(prefToTitleMap.get(key));
-                    screen.getPreference(i).setTitle(localizedString);
-                } catch (NoLocalizedTextException nle) {
-                    Log.w(TAG, "Unable to localize: " + prefToTitleMap.get(key));
-                }
-            }
-        }
+    @Override
+    protected String getTitle() {
+        return Localization.get("settings.main.title");
     }
 
-    public static void addBackButtonToActionBar(Activity activity) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
-            ActionBar actionBar = activity.getActionBar();
-            if (actionBar != null) {
-                actionBar.setDisplayShowHomeEnabled(true);
-                actionBar.setDisplayHomeAsUpEnabled(true);
-            }
-        }
+    @Override
+    protected Map<String, String> getPrefKeyAnalyticsEventMap() {
+        return prefKeyToAnalyticsEvent;
+    }
+
+    @Override
+    protected Map<String, String> getPrefKeyTitleMap() {
+        return keyToTitleMap;
+    }
+
+    @Override
+    protected int getPreferencesResource() {
+        return R.xml.commcare_preferences;
     }
 
-    private void setupButtons() {
+    @Override
+    protected boolean isAppLevelPreference() {
+        return true;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        setVisibilityOfUpdateOptionsPref();
```

Comment:
For this I think onStart() is a better place as it's related to visibility rather than any ui interaction and as UI becomes visible in OnStart() we should not defer it to onResume.